### DISCUSSION TITLE
Deploy including git repository to server

### DISCUSSION THREAD

**Discussion by Aquive** [deployphp/deployer#3353] [Help needed] [answered]
Simple question: is it possible to also deploy the git repository itself to the server?

**Comment by antonmedv** [comment]
Like bare repo?

**Comment by Aquive** [comment]
The repository of the project itself (I think in essence the .git/ folder?). The purpose is to be able to do `git status` on the server to see changed files.

**Reply by antonmedv** [reply]
Deployer already doing it)

**Comment by Aquive** [comment]
Thanks for your help! I have several projects where this is not the case. Do I need to do some configuration for it? Or how can I fix it?

**Comment by Aquive** [comment]
@antonmedv any ideas?

**Reply by antonmedv** [reply]
I don’t get the question. Try to use run() func directly.

**Comment by Aquive** [comment]
@antonmedv thanks for your reply.

Ok  steps to reproduce (for me).

- Deploy to server with deployer (`dep deploy`)
- Login to server with SSH
- Go to project directory
- Typ in `git status`

### Expected result:
```
app@e1a372088c46 ~/application (master)$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean`

```


### Actual result

```
app@app:~/public_html/application$ git status
fatal: not a git repository (or any parent up to mount point /)
```

This should make the issue clear, right?

And what do you mean by using `run()`?

Thanks for the help!

**Reply by antonmedv** [reply]
Simple write a task what will be executing run() commands. And use hooks to add it.

**Comment by Aquive** [comment]
@antonmedv can you give an example? I have a plain magento 2 recipe for now. Or is there a discord server for asking help maybe?

**Comment by Aquive** [comment]
@antonmedv this is what ChatGPT says. Is this correct? I will try it out and report here.

I apologize for the confusion. By default, Deployer does not deploy the Git repository to the server during the deployment process. Instead, it typically deploys the files from a specific branch or tag to the server, without including the Git repository itself.

However, if you specifically want to include the Git repository in the deployment, you can use a custom Deployer recipe that includes additional steps to copy the .git folder to the server along with the rest of the files. Here's an example of how you can modify your deploy.php file to include the Git repository:



```
// deploy.php

require 'recipe/magento2.php';

// ... (other deployment configuration)

// Custom task to include Git repository
task('deploy:git', function () {
    // Define the path to the .git folder
    $gitPath = '.git';
    
    // Define the path on the server where the .git folder should be copied
    $remoteGitPath = '{{release_path}}/.git';

    // Run the shell command to copy the .git folder to the server
    run("cp -R {{local_path}}/{$gitPath} {{release_path}}");

    // Run the shell command to remove the local .git folder after copying
    run("rm -rf {{local_path}}/{$gitPath}");

    // Run the shell command to change ownership and permissions of the .git folder on the server
    run("chown -R {{user}}:{{group}} {$remoteGitPath}");
    run("chmod -R 700 {$remoteGitPath}");
});

// Include the custom task in the deployment flow
after('deploy:release', 'deploy:git');
```

With this custom task added to your deploy.php file, the .git folder will be included in the deployment process and copied to the server along with the rest of the files. Please note that including the Git repository in the deployment may pose security risks, as it could potentially expose sensitive information, such as credentials and configuration files, to the server. Therefore, it's important to carefully consider the security implications and protect your Git repository and server accordingly.

**Reply by antonmedv** [reply]
Yes, you are correct. By default, Deployer does not deploy the Git repository to the server during the deployment process. Instead, it deploys the files from a specific branch or tag without including the Git repository itself.

If you want to include the Git repository in the deployment, you can use a custom Deployer recipe with additional steps to copy the .git folder to the server along with the rest of the files, as you've shown in the example deploy.php file.

Please note that including the Git repository in the deployment may pose security risks, as it could potentially expose sensitive information. It's essential to carefully consider the security implications and protect your Git repository and server accordingly.

**Reply by antonmedv** [reply]
(c) ChatGPT

**Comment by Aquive** [comment, marked as answer]
@antonmedv all the above didn't work. The solution was just changing below setting from archive to clone. I guess I couldn't make my issue clear enough, since the solution is very simple? Am I right?

`set('update_code_strategy', 'clone');`
